packages/jest-matchers/src/matchers.js

```diff
@@ -408,6 +408,76 @@ const matchers: MatchersObject = {
   },
 
   toMatchSnapshot,
+
+  toMatchObject(receivedObject: Object, expectedObject: Object) {
+
+    if (typeof receivedObject !== 'object' || receivedObject === null) {
+      throw new Error(
+        matcherHint('[.not].toMatch', 'object', 'expected') + '\n\n' +
+        `${RECEIVED_COLOR('string')} value must be an object.\n` +
+        printWithType('Received', JSON.stringify(receivedObject), printReceived),
+      );
+    }
+
+    if (typeof expectedObject !== 'object' || expectedObject === null) {
+      throw new Error(
+        matcherHint('[.not].toMatch', 'object', 'expected') + '\n\n' +
+        `${RECEIVED_COLOR('string')} value must be an object.\n` +
+        printWithType('Received', JSON.stringify(expectedObject), printReceived),
+      );
+    }
+
+    const compare = (expected: any, actual: any) => {
+
+      if (typeof actual !== typeof expected) {
+        return false;
+      }
+      if (typeof expected !== 'object' || expected === null) {
+        return expected === actual;
+      }
+
+      if (Array.isArray(expected)) {
+        if (!Array.isArray(actual)) {
+          return false;
+        }
+
+        return expected.every(exp => {
+          return actual.some(act => {
+            return compare(exp, act);
+          });
+        });
+      }
+
+      if (expected instanceof Date && actual instanceof Date) {
+        return expected.getTime() === actual.getTime();
+      }
+
+      return Object.keys(expected).every(key => {
+        const exp = expected[key];
+        const act = actual[key];
+        if (typeof exp === 'object' && exp !== null && act !== null) {
+          return compare(exp, act);
+        }
+        return act === exp;
+      });
+    };
+
+    const pass = compare(expectedObject, receivedObject);
+
+    const message = pass
+      ? () => matcherHint('.not.toMatchObject') +
+     `\n\nExpected value not to match object:\n` +
+     `  ${printExpected(JSON.stringify(expectedObject))}` +
```

Comment:
I think I have an idea. I will delete all the properties that are contained in the Received object but not contained in the matching object. I can then just use `jest-diff` without introducing any changes there! Expect an update with this soon!